core/src/main/java/org/sonarsource/sonarlint/core/notifications/SonarQubeNotifications.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * SonarLint Core - Implementation
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.notifications;
+
+import java.util.List;
+import java.util.Random;
+import java.util.Timer;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import org.sonarsource.sonarlint.core.client.api.common.NotificationConfiguration;
+import org.sonarsource.sonarlint.core.client.api.connected.ServerConfiguration;
+import org.sonarsource.sonarlint.core.client.api.notifications.SonarQubeNotificationListener;
+import org.sonarsource.sonarlint.core.container.connected.SonarLintWsClient;
+
+public class SonarQubeNotifications {
+  private static final int DELAY = 60_000;
+  private static final Object LOCK = new Object();
+
+  private static SonarQubeNotifications singleton;
+
+  private final List<NotificationConfiguration> configuredNotifications = new CopyOnWriteArrayList<>();
+
+  private Timer timer;
+  private NotificationTimerTask task;
+
+  SonarQubeNotifications(Timer timer, NotificationTimerTask task) {
+    this.timer = timer;
+    this.task = task;
+    this.timer.scheduleAtFixedRate(task, randomizeInitialDelay(), DELAY);
+  }
+
+  /**
+   * Generated a random number between DELAY*0.5 and DELAY*1.5
+   */
+  private static int randomizeInitialDelay() {
+    Random r = new Random();
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Use a cryptographically strong random number generator (RNG) like "java.security.SecureRandom" in place of this PRNG [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2245)